### PR TITLE
feat(bulk-update)!: replace bulk by bulk-update endpoint

### PR DESCRIPTION
**Pull request by xrutayisire**
BREAKING CHANGE: drop support for bulk endpoint

- replace bulk by bulk-update endpoint

## Types of changes

- [ ] Chore (a non-breaking change which is related to package maintenance)
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Description

- Update everything related to bulk to match new bulk-update endpoint

## Checklist:

- [ ] My change requires an update to the official documentation.
- [x] All [TSDoc](https://tsdoc.org) comments are up-to-date and new ones have been added where necessary.
- [x] All new and existing tests are passing.

### CODE REVIEW

File: src/client.ts
Comment (angeloashmore) [inline src/client.ts]: 💡 #idea: This phrasing gives the reader some direction:

> A custom type with published documents cannot be deleted. Delete all of the custom type's documents or remove the delete operation from the request before trying again.

If it's possible to know which custom type blocks the operation, maybe we could provide it in the message and as an instance property on the error (e.g. `customTypeID`):

> The "${customTypeID}" custom type has published documents and cannot be deleted. Unpublish all of its documents or remove the delete operation from the request before trying again.

File: src/client.ts
Comment (angeloashmore) [inline src/client.ts:611]: 🎉 #nice: The more specific error is a nice improvement. 🙂 And it's backward compatible!

File: src/errors.ts
Comment (angeloashmore) [inline src/errors.ts]: 💡 #idea: Aligning the name with the error would help with readability:

```diff
- type BulkUpdateTransactionErrorAPIResponse = {
+ type BulkUpdateHasExistingDocumentsErrorAPIResponse = {
```

(`BulkTransactionErrorAPIResponse` was improperly named.)

**Review by angeloashmore — COMMENTED** [review]
Looks good! I left a few notes about the new error. I'm not sure if it would require a lot of extra work, so I'll leave it up to you whether or not it is worth the effort.

File: src/client.ts
Comment (xrutayisire) [reply src/client.ts]: No, we don't know from the custom type API. That's why later on we are fetching the list of docs per custom types we delete :/
Maybe it should be something to improve on customtypes-api side, but not yet done

File: src/errors.ts
Comment (xrutayisire) [reply src/errors.ts]: Done

File: src/client.ts
Comment (angeloashmore) [reply src/client.ts]: In that case, would you be good with updating the error message to something like this?

> A custom type with published documents cannot be deleted. Delete all of the custom type's documents or remove the delete operation from the request before trying again.

**Review by angeloashmore — APPROVED** [review]
Looks good to merge!

[I left one suggestion](https://github.com/prismicio/prismic-custom-types-client/pull/15#discussion_r1621369546) about the new error message. I'll let you decide if it's better before merging + publishing.

File: src/client.ts
Comment (xrutayisire) [reply src/client.ts]: Done

**Pull request merged**
